### PR TITLE
add and use strtable object

### PR DESCRIPTION
**Pull request by r10s**
this pr adds a strtable helper object that can be used to handle the synchronous DC_GET_STRING requests.

a napi-function that actually wraps strtable_set_str() is missing, the other places where strtable has to  be called from module.c are added but not compiled or tested.

strtable.c as such is tested and should work as expected.

@ralphtheninja if you feel it is simpler to add the strtable objects as a global object instead of adding it to the context, feel free to do so.

@ralphtheninja this pr can only be merged usefully after https://github.com/deltachat/deltachat-core/pull/411 is merged.

### CODE REVIEW

**Comment by ralphtheninja** [comment]
I can update the submodule to v0.24.0 once https://github.com/deltachat/deltachat-core/pull/411 is merge and after that we can rebase this branch

File: src/strtable.c [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+	pthread_mutex_t mutex;
+	char*           str[MR_STR_COUNT];
+} strtable_t;
+
+
+strtable_t* strtable_new()
+{
+	strtable_t* strtable = calloc(1, sizeof(strtable_t));
+	if (strtable==NULL) {
+		exit(666);
```
Comment (ralphtheninja) [inline src/strtable.c]: :guitar:

**Comment by r10s** [comment]
merged https://github.com/deltachat/deltachat-core/pull/411 :)

**Comment by ralphtheninja** [comment]
@r10s Get some compile errors:

screenshot from 2018-10-30 01-46-43

**Comment by r10s** [comment]
@ralphtheninja sorry, was using the old prefix, still in my mind here and there :)

**Comment by ralphtheninja** [comment]
@r10s Seems to be working well. Just gonna add some docs and then we can merge it.

**Pull request merged**